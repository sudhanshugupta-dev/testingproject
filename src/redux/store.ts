import { configureStore } from '@reduxjs/toolkit';
import authReducer from './slices/authSlice';
import themeReducer from './slices/themeSlice';
import languageReducer from './slices/languageSlice';
import chatsReducer from './slices/chatsSlice';
import messagesReducer from './slices/messagesSlice';
import appBootstrapReducer from './slices/appBootstrap';

export const store = configureStore({
  reducer: {
    auth: authReducer,
    theme: themeReducer,
    language: languageReducer,
    chats: chatsReducer,
    messages: messagesReducer,
    app: appBootstrapReducer,
  },
  middleware: (getDefault) => getDefault({ serializableCheck: false }),
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
