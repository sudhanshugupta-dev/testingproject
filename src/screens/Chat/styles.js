import { StyleSheet } from 'react-native';
import { getThemeColors } from '../../themes';

// Create theme-aware styles function
export const createStyles = (mode = 'light') => {
  const colors = getThemeColors(mode);
  return StyleSheet.create({
    container: { 
      flex: 1 
    },

    searchBar: {
      padding: 10,
      borderBottomWidth: 1,
      borderBottomColor: colors.border,
    },
    searchInput: {
      borderRadius: 12,
      paddingHorizontal: 12,
      height: 40,
      fontSize: 16,
      backgroundColor: colors.card,
      color: colors.text,
    },

    row: {
      flexDirection: 'row',
      alignItems: 'center',
      padding: 14,
      borderBottomWidth: 1,
      borderBottomColor: colors.border,
    },
    name: { 
      fontWeight: '600', 
      fontSize: 16,
      color: colors.text,
    },
    last: { 
      marginTop: 4, 
      fontSize: 14,
      color: colors.textSecondary,
    },

    emptyContainer: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingBottom: 80,
    },
    emptyText: {
      marginTop: 10,
      fontSize: 16,
      fontWeight: '500',
      textAlign: 'center',
      paddingHorizontal: 20,
      color: colors.textSecondary,
    },
    unreadBadge: {
      minWidth: 22,
      height: 22,
      borderRadius: 11,
      alignItems: 'center',
      justifyContent: 'center',
      paddingHorizontal: 6,
      backgroundColor: colors.primary,
    },
    unreadText: {
      color: colors.background,
      fontSize: 12,
      fontWeight: 'bold',
    },
    timestamp: {
      fontSize: 12,
      fontWeight: '400',
    },
  });
};

// Backward compatibility - default light theme styles
export const styles = createStyles('light');
