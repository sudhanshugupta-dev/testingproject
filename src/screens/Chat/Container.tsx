import React, { useEffect, useState, useCallback } from 'react';
import {
  View,
  FlatList,
  Text,
  Pressable,
  StyleSheet,
  TextInput,
  ActivityIndicator,
} from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { AppDispatch, RootState } from '../../redux/store';
import { getChatList } from '../../redux/slices/chatsSlice';
import CustomAvatar from '../../components/CustomAvatar';
import { useNavigation } from '@react-navigation/native';
import LottieView from 'lottie-react-native';
import { useAppTheme } from '../../themes/useTheme';
import { useTranslation } from 'react-i18next';

const ChatContainer = () => {
  const dispatch = useDispatch<AppDispatch>();
  const { list, loading } = useSelector((s: RootState) => s.chats);
  const nav = useNavigation<any>();
  const [search, setSearch] = useState('');
  const { colors } = useAppTheme();
  const { t } = useTranslation();

  // âœ… Fetch chat list
  useEffect(() => {
    dispatch(getChatList());
  }, [dispatch]);

  // âœ… Filter chats by search
  const filteredList = list.filter(
    item =>
      item.name?.toLowerCase().includes(search.toLowerCase()) ||
      item.email?.toLowerCase().includes(search.toLowerCase()),
  );

  // âœ… Render each chat row
  const renderItem = useCallback(
    ({ item }: { item: any }) => (
      <Pressable
        style={[styles.row, { borderBottomColor: colors.text + '22', backgroundColor: colors.background }]}
        android_ripple={{ color: '#E5E7EB' }}
        onPress={() =>
          nav.navigate('ChatRoom', { friendId: item.id, friendName: item.name })
        }
      >
        <CustomAvatar name={item.name} />
        <View style={{ marginLeft: 12, flex: 1 }}>
          <Text style={[styles.name, { color: colors.text }]}>{item.name}</Text>
          <Text numberOfLines={1} style={[styles.last, { color: colors.text, opacity: 0.6 }]}>
            {item.lastMessage}
          </Text>
        </View>
      </Pressable>
    ),
    [nav, colors],
  );

  return (
    <View style={[styles.container, { backgroundColor: colors.background }]}>      
      <View style={[styles.searchBar, { backgroundColor: colors.background, borderBottomColor: colors.text + '22' }]}>
        <TextInput
          placeholder={t('chat.typeMessage')}
          value={search}
          onChangeText={setSearch}
          style={[styles.searchInput, { backgroundColor: colors.card, color: colors.text }]}
          placeholderTextColor="#9CA3AF"
        />
      </View>

      {loading ? (
        <ActivityIndicator size="large" color={colors.primary} style={{ flex: 1 }} />
      ) : filteredList.length === 0 ? (
        // ðŸŸ¢ Empty State with Lottie
        <View style={styles.emptyContainer}>
          <LottieView
            source={require('../../assets/Chatbot.json')}
            autoPlay
            loop
            style={{ width: 200, height: 200 }}
          />
          <Text style={[styles.emptyText, { color: colors.text, opacity: 0.7 }]}>
            No chats yet, start a conversation!
          </Text>
        </View>
      ) : (
        <FlatList
          data={filteredList}
          keyExtractor={i => i.id}
          refreshing={loading}
          onRefresh={() => dispatch(getChatList())}
          renderItem={renderItem}
          showsVerticalScrollIndicator={false}
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: { flex: 1 },

  searchBar: {
    padding: 10,
    borderBottomWidth: 1,
  },
  searchInput: {
    borderRadius: 12,
    paddingHorizontal: 12,
    height: 40,
    fontSize: 16,
  },

  row: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 14,
    borderBottomWidth: 1,
  },
  name: { fontWeight: '600', fontSize: 16 },
  last: { marginTop: 4, fontSize: 14 },

  emptyContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingBottom: 80,
  },
  emptyText: {
    marginTop: 10,
    fontSize: 16,
    fontWeight: '500',
  },
});

export default ChatContainer;
