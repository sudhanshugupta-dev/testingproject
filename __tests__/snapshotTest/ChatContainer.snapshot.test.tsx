import React from 'react';
import { Provider } from 'react-redux';
import { store } from '../../src/redux/store';
import Chat from '../../src/screens/Chat';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';

it('Chat screen renders', () => {
  const tree = render(
    <Provider store={store}>
      <NavigationContainer>
        <Chat />
      </NavigationContainer>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
}); 