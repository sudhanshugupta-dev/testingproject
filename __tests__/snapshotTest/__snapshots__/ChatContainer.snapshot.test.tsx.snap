// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat screen renders 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "borderBottomWidth": 1,
          "padding": 10,
        },
        {
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "#11111122",
        },
      ]
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="chat.typeMessage"
      placeholderTextColor="#9CA3AF"
      style={
        [
          {
            "borderRadius": 12,
            "fontSize": 16,
            "height": 40,
            "paddingHorizontal": 12,
          },
          {
            "backgroundColor": "#F3F4F6",
            "color": "#111111",
          },
        ]
      }
      value=""
    />
  </View>
  <ActivityIndicator
    color="#4F46E5"
    size="large"
    style={
      {
        "flex": 1,
      }
    }
  />
</View>
`;
